File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int sumNaturalNums = n*(n+1) / 2;
        int arraySum = 0;
        
        for (int val : nums) {
            arraySum += val;
        }

        return sumNaturalNums - arraySum;
    }
}

File: 0268-missing-number/0268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let n = nums.length;

    // calculate sum of first n natural nums
    let sumN = (n * (n + 1)) / 2;
    
    // calculate total sum in nums
    let sumNums = 0;
    for (let i = 0; i < n; i++) {
        sumNums += nums[i];
    }

    return sumN - sumNums;
};